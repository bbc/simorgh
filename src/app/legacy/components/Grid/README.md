# Grid

## Usage

When using an extended Grid component, e.g. `FrontPageGrid`, you can change the element that's rendered by using the `as` prop.
e.g.
`<Grid as="main" role="main>...</Grid>`
`<FrontPageGrid as="main" role="main>...</FrontPageGrid>`
