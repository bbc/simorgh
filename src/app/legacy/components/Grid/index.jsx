import React, { useContext } from 'react';
import styled from '@emotion/styled';
import { string, node, number } from 'prop-types';
import GRID from '#psammead/psammead-grid/src';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MAX,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MAX,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MAX,
  GEL_GROUP_5_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';
import {
  GEL_MARGIN_BELOW_400PX,
  GEL_MARGIN_ABOVE_400PX,
  GEL_SPACING_DBL,
} from '#psammead/gel-foundations/src/spacings';
import { ServiceContext } from '../../../contexts/ServiceContext';

const Grid = React.forwardRef((props, ref) => {
  const { dir } = useContext(ServiceContext);

  return <GRID dir={dir} {...props} ref={ref} />;
});

const fourOfSixColumnsMaxWidthGroup4 = `30rem`;
/* (group4ColWidth 6.75rem * 4) + (3 * 16px gutters) = 27rem + 3rem = 30rem */

const eightOfTwelveColumnsMaxWidthGroup5 = `30.6rem`;
/* (group5ColWidth 2.95rem * 8) + (7 * 16px gutters) = 23.6rem + 7rem = 30.6rem */

const fiveOfSixColumnsMaxWidthScaleable = `83.33%`;
// (5 / 6) * 100 = 83.3333.. = 83.33%

const fourOfSixColumnsMaxWidthScaleable = `66.67%`;
// (4 / 6) * 100 = 66.6666.. = 66.67%

export const gelGridMargin = `
  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_MARGIN_BELOW_400PX};
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    padding: 0 ${GEL_MARGIN_ABOVE_400PX};
  }
`;

const layoutGridItemSmall = ({ padding = {} }) => `
  ${gelGridMargin}

  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    ${padding.group2 ? `padding: 0 ${padding.group2}` : ''};
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
    max-width: ${fourOfSixColumnsMaxWidthScaleable};
  }
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    max-width: ${fiveOfSixColumnsMaxWidthScaleable};
    ${padding.group3 ? `padding: 0 ${padding.group3}` : ''};
  }
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
    max-width: ${fourOfSixColumnsMaxWidthGroup4};
  }
  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    max-width: ${eightOfTwelveColumnsMaxWidthGroup5};
  }

  @supports (display: grid) {
    max-width: initial;
  }
`;

export const GelPageGrid = styled(Grid)`
  width: 100%;
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
    margin: 0 auto;
    max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  }
  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    margin: 0 auto;
    max-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN};
  }
`;

/* The following components relate to Grid configuration and Grid styles used on the following page types:
 * STY,MAP,PGL,Front Page,IDX page
 */
const StyledCPSPageGrid = styled(Grid)`
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    margin: 0 auto;
    max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  }
`;

export const CPSPageGrid = ({ children, ...props }) => (
  <StyledCPSPageGrid
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 8,
      group5: 8,
    }}
    enableGelGutters
    {...props}
  >
    {children}
  </StyledCPSPageGrid>
);

export const GridWrapper = props => (
  <GelPageGrid
    {...props}
    enableGelGutters
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 8,
      group5: 20,
    }}
  />
);

export const GridItemSmall = styled(props => (
  <Grid
    {...props}
    item
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: 5,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 4,
      group3: 5,
      group4: 4,
      group5: 8,
    }}
  />
))`
  ${layoutGridItemSmall}
`;

export const GridItemMedium = ({ gridColumnStart, gridSpan, ...props }) => (
  <Grid
    {...props}
    item
    margins={{
      group0: true,
      group1: true,
      group2: true,
      group3: true,
      group4: false,
      group5: false,
    }}
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: gridColumnStart,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 5,
      group4: 5,
      group5: gridSpan,
    }}
  />
);

export const GridItemMediumNoMargin = ({
  gridColumnStart,
  gridSpan,
  ...props
}) => (
  <Grid
    {...props}
    item
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: gridColumnStart,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 5,
      group4: 5,
      group5: gridSpan,
    }}
  />
);

export const GridItemLarge = props => (
  <Grid
    {...props}
    item
    margins={{
      group0: true,
      group1: true,
      group2: true,
      group3: true,
      group4: false,
      group5: false,
    }}
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: 5,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 6,
      group5: 12,
    }}
  />
);

export const GridItemLargeNoMargin = props => (
  <Grid
    {...props}
    item
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: 5,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 6,
      group5: 12,
    }}
  />
);

// 1.
// The max-height must be 0 at Group 5 breakpoints so that
// the item does not force the following sibling item downwards.

const PopOutAtGroup5 = styled(GridItemMedium)`
  @supports (display: grid) {
    @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
      max-height: 0; /* [1] */
      padding-top: 0.25rem;
      margin: 0 ${GEL_SPACING_DBL};
    }
  }
`;
export const PopOutGridItemMedium = ({ children, ...props }) => (
  <PopOutAtGroup5 {...props}>{children}</PopOutAtGroup5>
);

GridItemLarge.propTypes = {
  as: string,
};

GridItemLarge.defaultProps = {
  as: undefined,
};

GridItemLargeNoMargin.propTypes = {
  as: string,
};

GridItemLargeNoMargin.defaultProps = {
  as: undefined,
};

GridWrapper.propTypes = {
  as: string,
};

GridWrapper.defaultProps = {
  as: undefined,
};

GridItemMediumNoMargin.propTypes = {
  as: string,
  gridColumnStart: number,
  gridSpan: number,
};

GridItemMediumNoMargin.defaultProps = {
  as: undefined,
};

GridItemMedium.propTypes = {
  as: string,
  gridColumnStart: number,
  gridSpan: number,
};

GridItemMedium.defaultProps = {
  as: undefined,
};

GridItemMediumNoMargin.defaultProps = {
  gridColumnStart: 5,
  gridSpan: 10,
};

GridItemMedium.defaultProps = {
  gridColumnStart: 5,
  gridSpan: 10,
};

CPSPageGrid.propTypes = {
  children: node.isRequired,
};

PopOutGridItemMedium.propTypes = {
  children: node.isRequired,
  gridColumnStart: number,
  gridSpan: number,
};

PopOutGridItemMedium.defaultProps = {
  gridColumnStart: 1,
  gridSpan: 4,
};

export default Grid;
