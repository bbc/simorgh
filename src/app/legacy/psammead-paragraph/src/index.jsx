import styled from '@emotion/styled';
import { shape, string, bool } from 'prop-types';
import { C_LUNAR, C_GREY_10 } from '#legacy/psammead-styles/src/colours';
import { GEL_SPACING_TRPL } from '#legacy/gel-foundations/src/spacings';
import { getBodyCopy } from '#legacy/gel-foundations/src/typography';
import { scriptPropType } from '#legacy/gel-foundations/src/prop-types';
import { getSansRegular } from '#legacy/psammead-styles/src/font-styles';

const Paragraph = styled.p`
  ${({ script }) => script && getBodyCopy(script)};
  ${({ service }) => getSansRegular(service)}
  color: ${({ darkMode }) => (darkMode ? C_LUNAR : C_GREY_10)};
  padding-bottom: ${GEL_SPACING_TRPL};
  margin: 0; /* Reset */
`;

Paragraph.propTypes = {
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
  darkMode: bool,
};

Paragraph.defaultProps = {
  darkMode: false,
};

export default Paragraph;
