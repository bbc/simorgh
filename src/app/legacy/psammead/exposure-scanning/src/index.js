import parseArgs from './args';
import { fetchPr, fetchIssue } from './fetch';
import { scanPr, scanIssue } from './scan';
import { patchPr, patchIssue } from './patch';

const redactPr = async (reqBody, regex) => {
  const pr = await fetchPr(reqBody);

  const scannedPr = await scanPr(pr, regex);

  if (scannedPr.foundMatch) {
    await patchPr(reqBody, scannedPr);
    throw new Error('Match found and patched.');
  }
};

const redactIssue = async (reqBody, regex) => {
  const issue = await fetchIssue(reqBody);

  const scannedIssue = await scanIssue(issue, regex);

  if (scannedIssue.foundMatch) {
    await patchIssue(reqBody, scannedIssue);
    throw new Error('Match found and patched.');
  }
};

const scanExposures = async () => {
  const { repo, flag, id, regexString } = parseArgs(process.argv);

  const regex = new RegExp(regexString, 'gi');

  const reqBody = {
    owner: 'bbc',
    repo,
    id,
  };

  await {
    '-pr': redactPr,
    '-issue': redactIssue,
  }[flag](reqBody, regex);

  return 'No matches found.';
};

export default scanExposures;
