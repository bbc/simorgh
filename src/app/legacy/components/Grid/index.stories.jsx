import React from 'react';
import Grid, { CPSPageGrid as CPSPageGridComponent } from '.';

const outerGridProps = {
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 6,
    group4: 6,
    group5: 6,
  },
};

const gridItemProps = {
  item: true,
  columns: {
    group0: 2,
    group1: 2,
    group2: 2,
    group3: 2,
    group4: 2,
    group5: 2,
  },
};

export default {
  Component: Grid,
  title: 'Components/Grid',
};

export const GridAsMain = () => (
  <Grid as="main" role="main" {...outerGridProps}>
    <Grid {...gridItemProps}>
      <p>Item with Main element as the parent grid</p>
    </Grid>
  </Grid>
);

export const CPSPageGrid = () => (
  <CPSPageGridComponent>
    <Grid {...gridItemProps}>
      <p>CPS page grid</p>
    </Grid>
  </CPSPageGridComponent>
);
