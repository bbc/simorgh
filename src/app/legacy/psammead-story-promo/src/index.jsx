import React from 'react';
import styled from '@emotion/styled';
import { node, bool, string, oneOf, shape } from 'prop-types';
import {
  GEL_SPACING_HLF,
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_SPACING_QUAD,
} from '#legacy/gel-foundations/src/spacings';
import {
  GEL_GROUP_1_SCREEN_WIDTH_MAX,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_2_SCREEN_WIDTH_MAX,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MAX,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
  GEL_GROUP_5_SCREEN_WIDTH_MIN,
} from '#legacy/gel-foundations/src/breakpoints';
import {
  getParagon,
  getLongPrimer,
  getPica,
  getDoublePica,
} from '#legacy/gel-foundations/src/typography';
import {
  C_EBON,
  C_GREY_10,
  C_METAL,
  C_SHADOW,
  C_WHITE,
} from '#legacy/psammead-styles/src/colours';
import {
  getSansRegular,
  getSerifMedium,
} from '#legacy/psammead-styles/src/font-styles';
import { scriptPropType } from '#legacy/gel-foundations/src/prop-types';
import { grid } from '#legacy/psammead-styles/src/detection';
import ImageGridItem from './ImageStyles';
import TextGridItem from './TextStyles';

const PROMO_TYPES = oneOf([
  'top',
  'regular',
  'leading',
  'topStories',
  'relatedContent',
]);

const wrapperTopStyles = `
  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    grid-template-columns: repeat(12, 1fr);
  }
`;

const wrapperRelatedContent = `
  background-color: ${C_WHITE};
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    display: block;
  };
`;

const wrapperTopStoriesStyle = ({ displayTimestamp }) => `
  background-color: ${C_WHITE};
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    display: block;
  };
  padding:  ${
    displayTimestamp
      ? GEL_SPACING
      : `${GEL_SPACING} ${GEL_SPACING} 0 ${GEL_SPACING}`
  };

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: ${
      displayTimestamp
        ? `${GEL_SPACING_DBL} ${GEL_SPACING} `
        : `${GEL_SPACING_DBL} ${GEL_SPACING} ${GEL_SPACING} ${GEL_SPACING}`
    };
  };

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    padding: ${
      displayTimestamp
        ? GEL_SPACING_DBL
        : `${GEL_SPACING_DBL} ${GEL_SPACING_DBL} ${GEL_SPACING} ${GEL_SPACING_DBL}`
    };
  };
`;

const wrapperRegularStyles = `
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    display: block;
  }
`;

const wrapperStyles = {
  top: wrapperTopStyles,
  regular: wrapperRegularStyles,
  topStories: wrapperTopStoriesStyle,
  relatedContent: wrapperRelatedContent,
  leading: '',
};

const StoryPromoWrapper = styled.div`
  position: relative; /* This is needed to contain the faux-block-link to the Story Promo */
  @supports (${grid}) {
    display: grid;
    grid-template-columns: repeat(6, 1fr);
    grid-column-gap: ${GEL_SPACING};

    @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
      grid-column-gap: ${GEL_SPACING_DBL};
    }
    ${({ promoType }) => wrapperStyles[promoType]}
  }
`;

const ImageContentsWrapper = styled.div`
  position: relative;
`;

const mediaIndicatorStylesTopLeading = `
  position: absolute;
  bottom: 0;
  > * {
    height: ${GEL_SPACING_QUAD};
    padding: ${GEL_SPACING} ${GEL_SPACING_HLF};
  }
`;

const mediaIndicatorStylesRegular = `
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    position: absolute;
    bottom: 0;
  }
  > * {
    @media (max-width: ${GEL_GROUP_1_SCREEN_WIDTH_MAX}) {
      height: 1.25rem;
      padding: ${GEL_SPACING_HLF} ${GEL_SPACING_HLF} 0;
    }

    height: ${GEL_SPACING_QUAD};
    padding: ${GEL_SPACING} ${GEL_SPACING_HLF};
  }
`;

/*
 These styles are to ensure we have the correct positioning
 & spacing of the Media Indicator over the Image in the Story Promo 
 */
const mediaIndicatorStyles = {
  top: mediaIndicatorStylesTopLeading,
  regular: mediaIndicatorStylesRegular,
  topStories: mediaIndicatorStylesRegular,
  relatedContent: mediaIndicatorStylesRegular,
  leading: mediaIndicatorStylesTopLeading,
};

const ImageOverlayWrapper = styled.div`
  ${({ promoType }) => mediaIndicatorStyles[promoType]}
`;

const headlineTopStoryTypography = script => getParagon(script);

const headlineRegularTypography = script => getPica(script);

const headlineLeadingStoryTypography = script => getDoublePica(script);

const headlineTypography = script => ({
  top: headlineTopStoryTypography(script),
  regular: headlineRegularTypography(script),
  topStories: headlineRegularTypography(script),
  relatedContent: headlineRegularTypography(script),
  leading: headlineLeadingStoryTypography(script),
});

const StandardColor = `
  color: ${C_EBON};
`;
const TopStoryColor = `
  color: ${C_GREY_10};
`;

const relatedContentStyle = `
  color: ${C_GREY_10};
  padding: 0;
`;
const headlineStyle = {
  top: StandardColor,
  regular: StandardColor,
  topStories: TopStoryColor,
  relatedContent: relatedContentStyle,
  leading: StandardColor,
};

export const Headline = styled.h3`
  margin: 0; /* Reset */
  padding-bottom: ${GEL_SPACING};
  ${({ service }) => getSerifMedium(service)}
  ${({ script, promoType }) => script && headlineTypography(script)[promoType]}
  ${({ promoHasImage }) =>
    !promoHasImage &&
    `display: inline-block;`} /* Needed for aligning Media Indicator with Headline */
  ${({ promoType }) => headlineStyle[promoType]}
`;

Headline.propTypes = {
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
  promoHasImage: bool,
  promoType: PROMO_TYPES,
};

Headline.defaultProps = {
  promoHasImage: true,
  promoType: 'regular',
};

export const RelatedContentBoxWrapper = styled.div`
  padding: ${GEL_SPACING};
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: ${GEL_SPACING_DBL};
  }
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    padding: ${GEL_SPACING} ${GEL_SPACING_DBL};
  }
`;

const summaryTopStoryStyles = `
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    display: none;
    visibility: hidden;
  }
`;

const summaryRegularStyles = `
  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
    display: none;
    visibility: hidden;
  }

  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    display: none;
    visibility: hidden;
  }
`;

const summaryStyles = {
  top: summaryTopStoryStyles,
  regular: summaryRegularStyles,
  topStories: summaryRegularStyles,
  relatedContent: summaryRegularStyles,
  leading: summaryRegularStyles,
};

export const Summary = styled.p`
  ${({ script }) => script && getLongPrimer(script)};
  ${({ service }) => getSansRegular(service)}
  color: ${C_SHADOW};
  margin: 0; /* Reset */
  padding-bottom: ${GEL_SPACING};

  ${({ promoHasImage }) => !promoHasImage && `padding-top: ${GEL_SPACING};`}

  ${({ promoType }) => summaryStyles[promoType]}
`;

Summary.propTypes = {
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
  promoHasImage: bool,
  promoType: PROMO_TYPES,
};

Summary.defaultProps = {
  promoHasImage: true,
  promoType: 'regular',
};

export const Link = styled.a`
  position: static;
  text-decoration: none;
  overflow-wrap: break-word;

  &:before {
    bottom: 0;
    content: '';
    left: 0;
    overflow: hidden;
    position: absolute;
    right: 0;
    top: 0;
    white-space: nowrap;
    z-index: 1;
  }

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  &:visited {
    color: ${C_METAL};
  }
  ${({ promoType }) => `${headlineStyle[promoType]}`}
`;

const StoryPromo = ({
  image,
  info,
  promoType,
  dir,
  displayImage,
  displayTimestamp,
  mediaIndicator,
  mediaIndicatorIsInline,
  ...props
}) => {
  const renderImage = displayImage && (
    <ImageGridItem dir={dir} promoType={promoType}>
      <ImageContentsWrapper>
        {image}
        {mediaIndicator && (
          <ImageOverlayWrapper
            mediaIndicatorIsInline={mediaIndicatorIsInline}
            promoType={promoType}
          >
            {mediaIndicator}
          </ImageOverlayWrapper>
        )}
      </ImageContentsWrapper>
    </ImageGridItem>
  );

  const renderText = (
    <TextGridItem promoType={promoType} dir={dir} displayImage={displayImage}>
      {!displayImage && mediaIndicator}
      {info}
    </TextGridItem>
  );

  return (
    <StoryPromoWrapper
      promoType={promoType}
      displayImage={displayImage}
      displayTimestamp={displayTimestamp}
      {...props}
    >
      {promoType === 'leading' ? (
        <>
          {renderText}
          {renderImage}
        </>
      ) : (
        <>
          {renderImage}
          {renderText}
        </>
      )}
    </StoryPromoWrapper>
  );
};

StoryPromo.propTypes = {
  dir: oneOf(['ltr', 'rtl']),
  image: node.isRequired,
  info: node.isRequired,
  promoType: PROMO_TYPES,
  displayImage: bool,
  displayTimestamp: bool,
  mediaIndicator: node,
  mediaIndicatorIsInline: bool,
};

StoryPromo.defaultProps = {
  dir: 'ltr',
  promoType: 'regular',
  displayImage: true,
  displayTimestamp: true,
  mediaIndicator: null,
  mediaIndicatorIsInline: false,
};

export default StoryPromo;
