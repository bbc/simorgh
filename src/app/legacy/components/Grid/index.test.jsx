import React from 'react';
import { shouldMatchSnapshot } from '#psammead/psammead-test-helpers/src';
import Grid, { CPSPageGrid } from '.';

const outerGridProps = {
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 6,
    group4: 6,
    group5: 6,
  },
};

const gridItemProps = {
  item: true,
  columns: {
    group0: 2,
    group1: 2,
    group2: 2,
    group3: 2,
    group4: 2,
    group5: 2,
  },
};

describe('Grid', () => {
  shouldMatchSnapshot(
    'should render Grid with columns',
    <Grid {...outerGridProps}>
      <Grid {...gridItemProps}>
        <p>item spanning 2 out of 6 columns</p>
      </Grid>
    </Grid>,
  );

  shouldMatchSnapshot(
    'should render CPSPageGrid',
    <CPSPageGrid>
      <Grid {...gridItemProps}>
        <p>CPS page grid</p>
      </Grid>
    </CPSPageGrid>,
  );

  shouldMatchSnapshot(
    'should render CPSPageGrid with additional props',
    <CPSPageGrid as="main" role="main">
      <Grid {...gridItemProps}>
        <p>CPS page grid</p>
      </Grid>
    </CPSPageGrid>,
  );
});
