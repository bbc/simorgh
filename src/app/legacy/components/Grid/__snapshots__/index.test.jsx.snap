// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render CPSPageGrid 1`] = `
@supports (display: grid) {
  .emotion-1 {
    display: grid;
    position: initial;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(8, 1fr);
      grid-column-end: span 8;
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 80rem) {
    .emotion-1 {
      grid-template-columns: repeat(8, 1fr);
      grid-column-end: span 8;
      grid-column-gap: 1rem;
    }
  }
}

@media (min-width: 63rem) {
  .emotion-1 {
    margin: 0 auto;
    max-width: 63rem;
  }
}

@supports (display: grid) {
  .emotion-3 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }
}

<div>
  <div
    class="emotion-0 emotion-1 emotion-2"
    dir="ltr"
  >
    <div
      class="emotion-3 emotion-2"
      dir="ltr"
    >
      <p>
        CPS page grid
      </p>
    </div>
  </div>
</div>
`;

exports[`Grid should render CPSPageGrid with additional props 1`] = `
@supports (display: grid) {
  .emotion-1 {
    display: grid;
    position: initial;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(8, 1fr);
      grid-column-end: span 8;
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 80rem) {
    .emotion-1 {
      grid-template-columns: repeat(8, 1fr);
      grid-column-end: span 8;
      grid-column-gap: 1rem;
    }
  }
}

@media (min-width: 63rem) {
  .emotion-1 {
    margin: 0 auto;
    max-width: 63rem;
  }
}

@supports (display: grid) {
  .emotion-3 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-3 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }
}

<div>
  <main
    class="emotion-0 emotion-1 emotion-2"
    dir="ltr"
    role="main"
  >
    <div
      class="emotion-3 emotion-2"
      dir="ltr"
    >
      <p>
        CPS page grid
      </p>
    </div>
  </main>
</div>
`;

exports[`Grid should render Grid with columns 1`] = `
@supports (display: grid) {
  .emotion-0 {
    display: grid;
    position: initial;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
    }
  }
}

@supports (display: grid) {
  .emotion-2 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-2 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-2 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-2 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-2 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-2 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-2 {
      grid-template-columns: repeat(2, 1fr);
      grid-column-end: span 2;
    }
  }
}

<div>
  <div
    class="emotion-0 emotion-1"
    dir="ltr"
  >
    <div
      class="emotion-2 emotion-1"
      dir="ltr"
    >
      <p>
        item spanning 2 out of 6 columns
      </p>
    </div>
  </div>
</div>
`;
